keps/sig-node/4191-split-image-filesystem/README.md

```diff
@@ -0,0 +1,899 @@
+sda1: [read-only layer]
+
+A goal is to allow kubelet to have separate disks for read-only layer and everything else could be stored on the same disk as Kubelet.
+
+In case of disk pressure on each filesystem, what is garbage collected/stored on the disk?
+
+Node Fileystem:
+
+- Pods
+- Logs
+- Containers
+- Ephemeral Storage
+
+Image Filesystem:
+
+- Images
+
+Node Filesystem should monitor storage for containers in addition to ephemeral storage.
+
+### Comment on Future Extensions
+
+There was a design decision for alpha in this KEP to go with support of a known customer request.  We foresee interest in the future for other use cases.
+So we want to comment on what work would be required to support these usecases.
+
+Separating images and containers into their own dedicated filesystems would require some extensions in CAdvisor, Stats Provider and the eviction manager.  
+Generally each filesystem will need to be tracked as part of the Stats Provider and that should be used to populate the Summary API.  
+The eviction manager would need to be aware of each filesystem that is added.
+
+Should CAdvisor monitor total capacity of each filesystem or should that be returned in CRI?
+
+Adding a new API for ImageFsStats that returns an array of both images and containers would allow for possible extension for CRI implementations to report file system totals and usage.
+
+Another extension can be multiple filesystems for images and containers.  
+The API allows for a list of filesystem usage per images and containers but there has been no work done to support
+this in other container runtimes.  
+
+### Risks and Mitigations
+
+By splitting the filesystem we allow more cases than what we currently support in Kubelet.
+To avoid bugs, we will validate on cases we don't currently support in Kubelet and return an error.
+
+The following cases will be validated and we will return an error if container runtime is set up for this:
+
+- Read-only layer and writable layer in separate locations on a separate disk from root filesystem.
+- Read-Only layer on root filesystem while Writeable layer on image filesystem.
+- Read-Only, Writable and root are all on separate disks.
+
+## Design Details
+
+### CRI
+
+CRIO and containerd return a single element in this case and Kubelet does not assume that there are multiple values in this array.  Regardless, we add an array to ImageFsInfoResponse.
+
+```golang
+// ImageService defines the public APIs for managing images.
+service ImageService {
+…
+    rpc ImageFsInfo(ImageFsInfoRequest) returns (ImageFsInfoResponse) {}
+}
+
+message ImageFsInfoResponse {
+    // Information of image filesystem(s).
+    repeated FilesystemUsage image_filesystems = 1;
+    // Information of container filesystem(s).
+    // This is an optional field if container and image
+    // storage are separated.
+    // Default will be to return this as empty.
+    repeated FilesystemUsage container_filesystems = 2;
```

Comment:
can you add diff syntax to show this section is added?

Reply:
done.